.github/workflows/update_4C_metadata_schema_file.yaml

```
@@ -0,0 +1,83 @@
+name: Update 4C metadata and schema file
+
+on:
+  schedule:
+    - cron: '0 5 * * *' # run each night at 5AM. Current nightly 4C test should be completed by then
+
+permissions:
+  contents: write
+
+jobs:
+  update-4C-metadata-schema-file:
+    name: Update 4C metadata and schema file
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Pull latest metadata file from 4C
+        uses: dawidd6/action-download-artifact@v9
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          workflow: nightly_tests.yml
+          workflow_conclusion: completed
+          name: clang18_build-metadata
+          path: clang18_build-metadata
+      - name: Pull latest schema file from 4C
+        uses: dawidd6/action-download-artifact@v9
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          workflow: nightly_tests.yml
+          workflow_conclusion: completed
+          name: clang18_build-schema
+          path: clang18_build-schema
+      - name: Check if metadata file has changed and overwrite old file if necessary
+        run: |
+          set -euo pipefail
+          hash_current=$(grep -v 'commit_hash' src/fourcipp/config/4C_metadata.yaml | sha256sum | awk '{print $1}')
+          hash_new=$(grep -v 'commit_hash' clang18_build-metadata/4C_metadata.yaml | sha256sum | awk '{print $1}')
+
+          echo "Current hash: $hash_current"
+          echo "New hash:     $hash_new"
```

```suggestion
          echo "Current metadata file hash: $hash_current"
          echo "New metadata file hash:     $hash_new"
```
In the hope we won't confuse it with the 4C hash

---

Good idea - changed it